vmm/src/vm.rs

```diff
@@ -548,6 +559,12 @@ impl Vm {
             )
             .map_err(Error::CpuManager)?;
 
+        #[cfg(feature = "igvm")]
+        let load_payload_handle = if snapshot.is_none() {
+            Self::load_payload_async(&memory_manager, &config, &cpu_manager)?
+        } else {
+            None
+        };
```

Comment:
Please add some comments here to explain why `load_payload_async()` needs to be pushed down for igvm (due to mshv limitations that will be addressed later).

Also, you missed a blank line below.